docs/source/module_configuration/staticsite.rst

```diff
@@ -111,3 +111,149 @@ Most of these options are not required, but are listed here for reference::
               build_output: dist  # overrides default directory of top-level path setting
         regions:
           - us-west-2
+
+`Authorization@Edge`
+--------------------
+
+Authorization, the function of specifying access rights to resources, is often
+required to help protect restricted content in web applications. In a
+`blog post <https://aws.amazon.com/blogs/networking-and-content-delivery/authorizationedge-how-to-use-lambdaedge-and-json-web-tokens-to-enhance-web-application-security/>`_
+written by Alex Tomic and Cameron Worrell an in-depth solution is described
+on how to accomplish this with CloudFront, `Lambda@Edge` functions, and a
+Cognito User Pool. A similar solution has been integrated directly into
+Runway, via the StaticSite module, to allow easy and repeatable idempotent
+launching of private static sites.
+
+The following diagram depicts a high-level overview of this solution.
+
+.. image:: ../images/staticsite/auth_at_edge/flow_diagram.png
+
+Here is how the solution works:
+
+    1. The viewer’s web browser is redirected to Amazon Cognito custom UI page to sign up and authenticate.
+    2. After authentication, Cognito generates and cryptographically signs a JWT then responds with a redirect containing the JWT embedded in the URL.
+    3. The viewer’s web browser extracts JWT from the URL and makes a request to private content (private/* path), adding Authorization request header with JWT.
+    4. Amazon CloudFront routes the request to the nearest AWS edge location. The CloudFront distribution’s private behavior is configured to launch a `Lambda@Edge` function on ViewerRequest event.
+    5. `Lambda@Edge` decodes the JWT and checks if the user belongs to the correct Cognito User Pool. It also verifies the cryptographic signature using the public RSA key for Cognito User Pool. Crypto verification ensures that JWT was created by the trusted party.
+    6. After passing all of the verification steps, `Lambda@Edge` strips out the Authorization header and allows the request to pass through to designated origin for CloudFront. In this case, the origin is the private content Amazon S3 bucket.
+    7. After receiving response from the origin S3 bucket, CloudFront sends the response back to the browser. The browser displays the data from the returned response.
+
+An example of a Auth@Edge static site configuration is as follows. All listed options are required::
+
+    deployments:
+      - modules:
+        - name: sample-app
+          path: sample-app
+          type: static
+          environments:
+            dev:
+              namespace: sample-app-dev
+              staticsite_auth_at_edge: true
+              staticsite_aae_user_pool_arn: arn:aws:cognito-idp:us-east-1:240134083525:userpool/us-east-1_cjVgcUyWV
```

Comment:
do the new options need to have a `staticsite_aae` prefix? I get that we need to keep `staticsite` for consistency but i don't think adding `aae` is necessary.

Reply:
Yeah I agree and will update them. It was an attempt to avoid naming conflicts initially, but now that we're through the bulk and there's no naming conflicts to deal with it was probably a poor choice in retrospect.